src/sonic-yang-models/yang-models/sonic-nat.yang

```diff
@@ -0,0 +1,281 @@
+                leaf nat_tcp_timeout {
+                    type uint32 {
+                        range "300..432000" {
+                            error-message "Invalid NAT TCP Timeout, valid range 300 sec to 432000 sec.";
+                            error-app-tag nat-tcp-timeout-invalid;
+                        }
+                    }
+                }
+                leaf nat_udp_timeout {
+                    type uint16 {
+                        range "120..600" {
+                            error-message "Invalid NAT UDP Timeout, valid range 120 sec to 600 sec.";
+                            error-app-tag nat-udp-timeout-invalid;
+                        }
+                    }
+                }
+            }
+        }
+
+        container NAT_POOL {
+            list NAT_POOL_LIST {
+                key "pool_name";
+                max-elements 16;
+
+                ext:key-regex-configdb-to-yang "^([a-zA-Z0-9_-]+)$";
+
+                ext:key-regex-yang-to-configdb "name";
+
+                leaf pool_name {
+                    type string {
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})';
+                        length 1..32 {
+                            error-app-tag pool-name-invalid-length;
+                        }
+                    }
+                }
+
+                leaf nat_ip {
+                    type ip-address;
+                }
+
+                leaf nat_port {
+                    type string {
+                        pattern "[0-9]{1,5}(-)[0-9]{1,5}";
+                    }
+                }
+            }
+        }
+
+        container NAT_BINDINGS {
+
+            list NAT_BINDINGS_LIST {
+
+                key "name";
+                max-elements 16;
+
+		ext:key-regex-configdb-to-yang "^([a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,15}))$";
+
+		ext:key-regex-yang-to-configdb "name";
+
+                leaf name {
+                    type string {
+
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})';
+                        length 1..32 {
+                            error-app-tag binding-name-invalid-length;
+                        }
+                    }
+                }
+
+                leaf nat_pool {
+                    mandatory true;
+                    type leafref {
+                        path "../../../NAT_POOL/NAT_POOL_LIST/pool_name";
+                    }
+                }
+
+                leaf access_list {
+                    type leafref {
+                        path "/sacl:sonic-acl/sacl:ACL_TABLE/sacl:ACL_TABLE_LIST/sacl:ACL_TABLE_NAME";
+                    } 
+                }
+
+                leaf nat_type {
+                    type nat-type;
+                    default snat;
+                }
+                uses sonic-twice-nat-id;
+            }
+        }
+    }
+
+    rpc clear_nat {
+        input {
+            leaf nat-param {
+                type nat-clear-param;
+                description
```

Comment:
Not sure we have any use-case with rpc in the community today, if not required, can this be removed?

Reply:
Will remove the rpc section.